### PR TITLE
document require_one_based_indexing

### PR DESCRIPTION
**Pull request by jw3126**
I think this function is a handy little helper and unlikely to change, so why not make it public API?

Edit: I just realized, that `Base.require_one_based_indexing` is already endorsed in the [docs](https://docs.julialang.org/en/v1/devdocs/offset-arrays/#man-custom-indices). So it is already public API.

### CODE REVIEW

File: base/abstractarray.jl [hunk 1]
```diff
@@ -101,12 +101,20 @@ end
 
 Return `true` if the indices of `A` start with something other than 1 along any axis.
 If multiple arguments are passed, equivalent to `has_offset_axes(A) | has_offset_axes(B) | ...`.
+See also [`require_one_based_indexing`](@ref)
```
Comment (johnnychen94) [inline base/abstractarray.jl:104]:
```suggestion
See also [`require_one_based_indexing`](@ref).
```

File: base/abstractarray.jl [hunk 2]
```diff
@@ -101,12 +101,20 @@ end
...
 has_offset_axes(A) = _tuple_any(x->Int(first(x))::Int != 1, axes(A))
 has_offset_axes(A::AbstractVector) = Int(firstindex(A))::Int != 1 # improve performance of a common case (ranges)
 has_offset_axes(A...) = _tuple_any(has_offset_axes, A)
 has_offset_axes(::Colon) = false
 
+"""
+    require_one_based_indexing(A::AbstractArray)
+    require_one_based_indexing(A,B...)
+
+Throw an `ArgumentError` if the indices of any argument start with something other than one along any axis.
```
Comment (johnnychen94) [inline base/abstractarray.jl:115]: I believe this requires Julia at least 1.1.0.

Comment (jw3126) [reply base/abstractarray.jl:115]: Oh, thanks for catching that. I will check and add an appropriate compat entry.

Comment (Moelf) [reply base/abstractarray.jl:115]:
```suggestion
Throw an `ArgumentError` if the indices of any argument start with something other than `1` along any axis.
```

**Comment by jw3126** [comment]
Oh nice thanks for merging @KristofferC

**Comment by KristofferC** [comment]
Hehe, sorry for late response.

**Pull request merged**